cpp/include/cudf/detail/utilities/device_operators.cuh

```diff
@@ -87,26 +86,12 @@ struct DeviceMin {
     return std::numeric_limits<T>::max();
   }
 
-  // @brief min op specialized for string_view (with null string_view as identity)
-  // because min op identity should const largest string, which does not exist
-  template <typename T,
-            typename std::enable_if_t<std::is_same<T, cudf::string_view>::value>* = nullptr>
-  CUDA_HOST_DEVICE_CALLABLE T operator()(const T& lhs, const T& rhs) {
-    if (lhs.is_null())
-      return rhs;
-    else if (rhs.is_null())
-      return lhs;
-    else
-      return std::min(lhs, rhs);
-    //return lhs <= rhs ? lhs : rhs;
-  }
-
   // @brief identity specialized for string_view 
-  // because string_view constructor requires 2 arguments
+  // (sentinel value which is the highest possible valid UTF-8 encoded character)
   template <typename T,
             typename std::enable_if_t<std::is_same<T, cudf::string_view>::value>* = nullptr>
   static constexpr T identity() {
-    return T{nullptr, 0};
+    return string_scalar("\xF7\xBF\xBF\xBF").value();
```

Comment:
This should be a `string_view`, should it not? Otherwise this is causing a device allocation and memcopy.

Reply:
`string_scalar::value()` returns `string_view`. (recently changed)
(`string_scalar::to_string()` give std::string)

Reply:
But this is the constructor that is being called: https://github.com/rapidsai/cudf/blob/6f81a5ccd21abb245fba869045cd63a4cd6250cc/cpp/include/cudf/scalar/scalar.hpp#L221

Which is going to incur a device memory allocation/H2D copy.

Reply:
Yes. I didn't see another way. The pointer is accessed in kernel during comparison. 

Alternatively: The sentinel string could be made char* device constant and `string_view(const_array, 4)` can be returned.

Reply:
I believe you should just be able to do something like:
```
constexpr string_view sentinel{"\xF7\xBF\xBF\xBF",4};
```

And if that doesn't work, then I think this should:
```
__device__
const string_view sentinel{"\xF7\xBF\xBF\xBF",4};
```

Reply:
both of this does not work.
`a constexpr variable must have a literal type or a reference type.`
` error : dynamic initialization is not supported for a __device__ variable`
https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#device-memory-specifiers

The existing code works without any error.

Reply:
What about your idea here?

> Alternatively: The sentinel string could be made char* device constant and string_view(const_array, 4) can be returned.

I think that should work?

Reply:
Yes. That works. 
(It took last couple of days to make it work; cuda programming guide c++ language support section helped.)

Also, having a global variable `string_scalar sentinel("\xF7\xBF\xBF\xBF");` and using `return sentinel.value();` in constexpr identity also works. But this may fail when used with multiple GPUs. so i skipped it.

Reply:
@jrhemstad  null string_view for both min and max replaced with sentinel.